beacon_node/store/src/iter.rs

```diff
@@ -261,6 +262,119 @@ fn slot_of_prev_restore_point<E: EthSpec>(current_slot: Slot) -> Slot {
     (current_slot - 1) / slots_per_historical_root * slots_per_historical_root
 }
 
+pub struct SlotBlockStateIterator<E: EthSpec, S> {
```

Comment:
I had a go at rewriting `prune_abandoned_forks` to not require the skipped slot information, and I think I got it to work. I think this is preferable because it means we don't need to introduce a new iterator type that provides similar functionality to the existing ones (we already have _a lot_ of iterators). Let me know what you think https://github.com/michaelsproul/lighthouse/commit/180a3812468d54332a353326820003b7b51e22be (you could cherry-pick that commit to this branch)

Reply:
And in future if we do really really need skip slot information, I think the simplest way to construct it would be with something like [`tuple_windows`](https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.tuple_windows) on a `BlockRootsIterator`.

Reply:
👍